### PR TITLE
Support Added Protocol Conformance to CoreBluetooth classes

### PR DESCRIPTION
**Pull request by cctile**
This PR adds support for adding protocol conformance to existing CoreBluetooth classes to support dependency injection. 

This work was pair-programmed with @paul-tile

### CODE REVIEW

File: DeannaMac/DEMAppDelegate.m [hunk 1]
```diff
@@ -156,41 +157,52 @@ - (BOOL)selectionShouldChangeInTableView:(NSTableView *)tableView {
...
+        YMSCBPeripheral *yp = [centralManager findPeripheral:yPeripheral];
+        yp.delegate = strongThis;
         
-        if (pvc.sensorTag == yp) {
-            pvc.rssiLabel.stringValue = [NSString stringWithFormat:@"%d", [RSSI intValue]];
+        if (self.oldCount == 0) {
+            self.oldCount = (int)centralManager.count;
+            [strongThis.peripheralTableView reloadData];
+        } else {
+            if (centralManager.count != self.oldCount) {
```
Comment (paul-tile) [inline DeannaMac/DEMAppDelegate.m]: strongThis.oldCount

File: DeannaMac/DEMAppDelegate.m [hunk 2]
```diff
@@ -156,41 +157,52 @@ - (BOOL)selectionShouldChangeInTableView:(NSTableView *)tableView {
...
+                self.oldCount = (int)centralManager.count;
```
Comment (paul-tile) [inline DeannaMac/DEMAppDelegate.m]: strongThis.oldCount

File: Deanna/Services/SensorTag/Services/DEABaseService.m [hunk 3]
```diff
@@ -51,21 +51,21 @@ - (instancetype)initWithName:(NSString *)oName
...
-    
-    yc = [[YMSCBCharacteristic alloc] initWithName:cname
-                                            parent:self.parent
-                                              uuid:uuid
-                                            offset:addrOffset];
-    
-    self.characteristicDict[cname] = yc;
-}
-
+//- (void)addCharacteristic:(NSString *)cname withOffset:(int)addrOffset {
```
Comment (paul-tile) [inline Deanna/Services/SensorTag/Services/DEABaseService.m]: Should we delete this commented out code now?

File: YmsCoreBluetooth/YMSCBCentralManager.m [hunk 4]
```diff
@@ -20,7 +20,9 @@
 #import "YMSCBPeripheral.h"
 #import "YMSCBService.h"
 #import "YMSCBCharacteristic.h"
-#import "YMSCBNativeCentralManager.h"
+#import "YMSCBNativeInterfaces.h"
+
+//#import "YMSCBNativeCentralManager.h"
```
Comment (paul-tile) [inline YmsCoreBluetooth/YMSCBCentralManager.m]: delete commented line

File: YmsCoreBluetooth/YMSCBCentralManager.m [hunk 5]
```diff
@@ -52,7 +54,8 @@ - (nullable instancetype)initWithDelegate:(nullable id<YMSCBCentralManagerDelega
         _queue = queue;
 
         // TODO: conditional compile based on environment
-        _centralInterface = [[YMSCBNativeCentralManager alloc] initWithDelegate:self queue:queue options:options];
+        //_centralInterface = [[YMSCBNativeCentralManager alloc] initWithDelegate:self queue:queue options:options];
```
Comment (paul-tile) [inline YmsCoreBluetooth/YMSCBCentralManager.m]: delete commented line

File: DeannaMac/DEMAppDelegate.m (hunk 1 above)
Comment (cctile) [inline DeannaMac/DEMAppDelegate.m]: Got it.

File: Deanna/Services/SensorTag/Services/DEABaseService.m (hunk 3 above)
Comment (cctile) [inline Deanna/Services/SensorTag/Services/DEABaseService.m]: Started working on this ad-hoc. Will need a future PR to scrub old documentation.

**Pull request merged**